src/views/TimersView.jsx

```diff
@@ -0,0 +1,129 @@
+import Stopwatch from '../components/timers/Stopwatch';
+import Countdown from '../components/timers/Countdown';
+import XY from '../components/timers/XY';
+import Tabata from '../components/timers/Tabata';
+import TimerList from '../components/timers/TimerList';
+import DisplayTotalWorkoutTime from '../components/generic/DisplayTotalWorkoutTime';
+import { TimerContext } from '../AppContext';
+import { updateUrlFromTimers } from '../utils/helpers';
+import { calculateTotalWorkoutTime, calculateRemainingTime } from '../utils/workoutTime';
+import './TimersView.css';
+
+const TimersView = () => {
+  const { 
+    timers, 
+    appTimerIndex,
+    stopwatchTime,
+    countdownTime,
+    xyTime,
+    tabataCountdown,
+    tabataRestdown,
+    stopwatchIsRunning,
+    countdownIsRunning,
+    xyIsRunning,
+    tabataIsRunning
+  } = useContext(TimerContext);
+  
+  const [totalWorkoutTime, setTotalWorkoutTime] = useState(0);
+  const [remainingTime, setRemainingTime] = useState(0);
+
+  const defaultTimers = [
+    { title: 'Stopwatch', C: <Stopwatch controls={true} index={-1} /> },
+    { title: 'Countdown', C: <Countdown controls={true} index={-1} /> },
+    { title: 'XY', C: <XY controls={true} index={-1} /> },
+    { title: 'Tabata', C: <Tabata controls={true} index={-1} /> },
+  ];
+
+  useEffect(() => {
+    const total = calculateTotalWorkoutTime(timers);
+    setTotalWorkoutTime(total);
+    setRemainingTime(total);
+  }, [timers]);
+
+  useEffect(() => {
+    if (appTimerIndex >= 0) {
+      const currentTimer = timers[appTimerIndex];
+      let currentTime = 0;
+
+      if (currentTimer) {
+        switch (currentTimer.title) {
+          case 'Stopwatch':
+            currentTime = stopwatchIsRunning ? 
+              (currentTimer.data?.targetValue || 0) - stopwatchTime : 0;
+            break;
+          case 'Countdown':
+            currentTime = countdownIsRunning ? countdownTime : 0;
+            break;
+          case 'XY':
+            currentTime = xyIsRunning ? xyTime : 0;
+            break;
+          case 'Tabata':
+            currentTime = tabataIsRunning ? 
+              (tabataCountdown + tabataRestdown) : 0;
+            break;
+        }
+      }
+
+      const remaining = calculateRemainingTime(timers, appTimerIndex, currentTime);
+      setRemainingTime(remaining);
+    }
+  }, [
+    appTimerIndex,
+    stopwatchTime,
+    countdownTime,
+    xyTime,
+    tabataCountdown,
+    tabataRestdown
+  ]);
+
+  const handleReorder = (sourceIndex, destinationIndex) => {
+    const reorderedTimers = Array.from(timers);
+    const [removed] = reorderedTimers.splice(sourceIndex, 1);
+    reorderedTimers.splice(destinationIndex, 0, removed);
+    
+    const updatedTimers = reorderedTimers.map((timer, index) => ({
+      ...timer,
+      index,
+      component: React.cloneElement(timer.component, { index })
+    }));
+    
+    updateUrlFromTimers(updatedTimers);
+  };
+
+  const handleRemove = (index) => {
```

Comment:
This doesn't seem to work as expected - it does not remove the timer from my workout list. If you're only updating the URL state with this handler, the app state needs to stay in sync with the url state. Otherwise, you need to update the state as well. You could just use the deleteTimer function you defined in the context